File: contests/leetcode-biweekly169/q2.py
# problem:  https://leetcode.cn/contest/biweekly-contest-169/problems/count-subarrays-with-majority-element-i/
from typing import *


class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        n = len(nums)
        times = [0] * (n + 1)
        for i in range(0, n):
            if nums[i] == target:
                times[i + 1] = times[i] + 1
            else:
                times[i + 1] = times[i]
        res = 0
        for i in range(1, n + 1):
            for j in range(i, n + 1):
                if times[j] - times[i - 1] > (j - i + 1) / 2:
                    res += 1
        return res


def main():
    test = Solution()
    print(test.countMajoritySubarrays([1, 2, 2, 3], target=2))
    print(test.countMajoritySubarrays([1, 2, 2, 3], target=2))


if __name__ == "__main__":
    main()

#
# Created by: ASUS At: 11/8/2025 10:35 PM
#

File: contests/leetcode-weekly503/q2.cpp
// problem: https://leetcode.cn/contest/weekly-contest-503/problems/password-strength/
#include <algorithm>
#include <iostream>
#include <ranges>
#include <unordered_set>
#include <vector>
#include "../utils/vector_helper.h"


using namespace std;

class Solution {
public:
    int passwordStrength(string password) {
        unordered_set<char> pass(password.begin(), password.end());
        int res = 0;
        for (const auto &c: pass) {
            if ('a' <= c and c <= 'z') {
                res += 1;
            } else if ('A' <= c and c <= 'Z') {
                res += 2;
            } else if ('0' <= c and c <= '9') {
                res += 3;
            } else if (c == '!' or c == '@' or c == '#' or c == '$') {
                res += 5;
            }
        }
        return res;
    }
};

int main(int argc, char *argv[]) {
    Solution test;
    cout << test.passwordStrength("aA1!") << endl;
    cout << test.passwordStrength("bbB11#") << endl;
}

//
// Created By ASUS At 2026-05-24 10:42
//
